pgvector/runtime/src/main/java/io/quarkiverse/langchain4j/pgvector/PgVectorEmbeddingStore.java

```diff
@@ -0,0 +1,289 @@
+        return ids;
+    }
+
+    /**
+     * Adds multiple embeddings and their corresponding contents that have been embedded to the store.
+     *
+     * @param embeddings A list of embeddings to be added to the store.
+     * @param embedded A list of original contents that were embedded.
+     * @return A list of auto-generated IDs associated with the added embeddings.
+     */
+    @Override
+    public List<String> addAll(List<Embedding> embeddings, List<TextSegment> embedded) {
+        List<String> ids = embeddings.stream().map(ignored -> randomUUID()).collect(toList());
+        addAllInternal(ids, embeddings, embedded);
+        return ids;
+    }
+
+    /**
+     * Finds the most relevant (closest in space) embeddings to the provided reference embedding.
+     *
+     * @param referenceEmbedding The embedding used as a reference. Returned embeddings should be relevant (closest) to this
+     *        one.
+     * @param maxResults The maximum number of embeddings to be returned.
+     * @param minScore The minimum relevance score, ranging from 0 to 1 (inclusive).
+     *        Only embeddings with a score of this value or higher will be returned.
+     * @return A list of embedding matches.
+     *         Each embedding match includes a relevance score (derivative of cosine distance),
+     *         ranging from 0 (not relevant) to 1 (highly relevant).
+     */
+    @Override
+    public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, int maxResults, double minScore) {
+        List<EmbeddingMatch<TextSegment>> result = new ArrayList<>();
+        try (Connection connection = setupConnection()) {
+            String referenceVector = Arrays.toString(referenceEmbedding.vector());
+            String query = String.format(
+                    "WITH temp AS (SELECT (2 - (embedding <=> '%s')) / 2 AS score, embedding_id, embedding, text, metadata FROM %s) SELECT * FROM temp WHERE score >= %s ORDER BY score desc LIMIT %s;",
+                    referenceVector, table, minScore, maxResults);
+            PreparedStatement selectStmt = connection.prepareStatement(query);
+
+            ResultSet resultSet = selectStmt.executeQuery();
+            while (resultSet.next()) {
+                double score = resultSet.getDouble("score");
+                String embeddingId = resultSet.getString("embedding_id");
+
+                PGvector vector = (PGvector) resultSet.getObject("embedding");
+                Embedding embedding = new Embedding(vector.toArray());
+
+                String text = resultSet.getString("text");
+                TextSegment textSegment = null;
+                if (isNotNullOrBlank(text)) {
+                    String metadataJson = Optional.ofNullable(resultSet.getString("metadata")).orElse("{}");
+
+                    // Replace TypeToken with TypeReference in Jackson
+                    TypeReference<Map<String, String>> typeReference = new TypeReference<Map<String, String>>() {
```

Comment:
probably a good idea to keep it around in a constant.

Reply:
Can you elaborate a bit on this ?

Reply:
turn it into a static final variable :)